src/lab_05/lab_05_ex_01.java

```diff
@@ -0,0 +1,44 @@
+package lab_05;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.Scanner;
+import java.security.MessageDigest;
+
+public class lab_05_ex_01 {
+    public static void msg() throws NoSuchAlgorithmException {
+        Scanner scanner = new Scanner(System.in);
+        System.out.print("Enter your password: ");
+        String password = scanner.nextLine();
+
+        if (password.length() < 8) {
+            System.out.println("Invalid password: Password must be at least 8 characters long.");
+            System.exit(0);
+        } else if (password.contains(" ")) {
+            System.out.println("Invalid password: Password must not contain any spaces.");
+            System.exit(0);
+        } else if (!Character.isUpperCase(password.charAt(0))) {
+            System.out.println("Invalid password: Password must begin with an uppercase letter.");
+            System.exit(0);
+        } else if (!password.matches(".*[#$%].*")) {
```

Comment:
Nu sunt toate caracterele speciale (**!@^ &** whatever else there is). In cerinta era doar un exemplu